hcloud/firewall.go

```diff
@@ -70,9 +70,24 @@ const (
 
 // FirewallResource represents a resource to apply the new Firewall on.
 type FirewallResource struct {
-	Type          FirewallResourceType
-	Server        *FirewallResourceServer
-	LabelSelector *FirewallResourceLabelSelector
+	Type               FirewallResourceType
+	Server             *FirewallResourceServer
+	LabelSelector      *FirewallResourceLabelSelector
+	AppliedToResources []FirewallAppliedResource
+}
+
+// FirewallAppliedResourceType specifies the type of resource that the Firewall is applied to.
+type FirewallAppliedResourceType string
+
+const (
+	// FirewallAppliedResourceTypeServer specifies a Server that a Firewall is applied to.
+	FirewallAppliedResourceTypeServer FirewallAppliedResourceType = "server"
+)
```

Comment:
Not sure if we actually need this. Is there a reason we should not use `FirewallResourceTypeServer` instead?